libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -79,6 +83,7 @@ proc replenishFanout(g: GossipSub, topic: string) {.async.} =
     if topic in g.gossipsub:
       for p in g.gossipsub[topic]:
         if not g.fanout[topic].containsOrIncl(p):
+          libp2p_gossipsub_peers_per_topic_fanout.inc(labelValues = [topic])
```

Comment:
if we have the count available here, better use `set` than `inc/dec` to avoid counting errors

Reply:
Altho containsOrIncl should guard from errors (the `not` is confusing)

Reply:
well, less risk with one canonical source